src/box/vy_scheduler.c

```diff
@@ -2073,6 +2073,12 @@ vy_scheduler_f(va_list va)
 		fiber_check_gc();
 		vy_scheduler_complete_tasks(scheduler, &tasks_done,
 					    &tasks_failed);
+		/*
+		 * Tasks completion may yield thus check fiber cancel
+		 * status before any wait.
+		 */
+		if (fiber_is_cancelled())
+			break;
```

Comment:
Let's remove the check in the `while` condition then. It doesn't look like we need it anymore.

Reply:
Done.

```diff
diff --git a/src/box/vy_scheduler.c b/src/box/vy_scheduler.c
index 29374c656..c28bd1324 100644
--- a/src/box/vy_scheduler.c
+++ b/src/box/vy_scheduler.c
@@ -2066,7 +2066,7 @@ vy_scheduler_f(va_list va)
 {
        struct vy_scheduler *scheduler = va_arg(va, struct vy_scheduler *);

-       while (!fiber_is_cancelled()) {
+       while (true) {
                struct vy_task *task;
                int tasks_done, tasks_failed;
```